internal/generator/vector/output/splunk/splunk.go

```diff
@@ -0,0 +1,129 @@
+package splunk
+
+import (
+	"fmt"
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	"github.com/openshift/cluster-logging-operator/internal/generator/vector/output/security"
+	"strings"
+
+	logging "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	. "github.com/openshift/cluster-logging-operator/internal/generator"
+	genhelper "github.com/openshift/cluster-logging-operator/internal/generator/helpers"
+	. "github.com/openshift/cluster-logging-operator/internal/generator/vector/elements"
+	vectorhelpers "github.com/openshift/cluster-logging-operator/internal/generator/vector/helpers"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	splunkEncodingJson = fmt.Sprintf("%q", "json")
+)
+
+type Splunk struct {
+	ComponentID  string
+	Inputs       string
+	Endpoint     string
+	DefaultToken string
+}
+
+func (s Splunk) Name() string {
+	return "SplunkVectorTemplate"
+}
+
+func (s Splunk) Template() string {
+	return `{{define "` + s.Name() + `" -}}
+[sinks.{{.ComponentID}}]
+type = "splunk_hec"
+inputs = {{.Inputs}}
+endpoint = "{{.Endpoint}}"
+compression = "none"
+default_token = "{{.DefaultToken}}"
+{{end}}`
+}
+
+type SplunkEncoding struct {
+	ComponentID string
+	Codec       string
+}
+
+func (se SplunkEncoding) Name() string {
+	return "splunkEncoding"
+}
+
+func (se SplunkEncoding) Template() string {
+	return `{{define "` + se.Name() + `" -}}
+[sinks.{{.ComponentID}}.encoding]
+codec = {{.Codec}}
+{{end}}`
+}
+
+func Conf(o logging.OutputSpec, inputs []string, secret *corev1.Secret, op Options) []Element {
+	if genhelper.IsDebugOutput(op) {
+		return []Element{
+			Debug(strings.ToLower(vectorhelpers.Replacer.Replace(o.Name)), vectorhelpers.MakeInputs(inputs...)),
+		}
+	}
+	return MergeElements(
+		[]Element{
+			Output(o, inputs, secret, op),
+			Encoding(o),
+		},
+		TLSConf(o, secret),
+	)
+}
+
+func Output(o logging.OutputSpec, inputs []string, secret *corev1.Secret, op Options) Element {
+	return Splunk{
+		ComponentID:  strings.ToLower(vectorhelpers.Replacer.Replace(o.Name)),
+		Inputs:       vectorhelpers.MakeInputs(inputs...),
+		Endpoint:     o.URL,
+		DefaultToken: security.GetFromSecret(secret, "hecToken"),
+	}
+}
+
+func Encoding(o logging.OutputSpec) Element {
+	return SplunkEncoding{
+		ComponentID: strings.ToLower(vectorhelpers.Replacer.Replace(o.Name)),
+		Codec:       splunkEncodingJson,
+	}
+}
+
+func TLSConf(o logging.OutputSpec, secret *corev1.Secret) []Element {
+	conf := []Element{}
+	if o.Secret != nil {
+		hasTLS := false
+		conf = append(conf, security.TLSConf{
+			ComponentID:        strings.ToLower(vectorhelpers.Replacer.Replace(o.Name)),
+			InsecureSkipVerify: o.TLS != nil && o.TLS.InsecureSkipVerify,
+		})
+
+		if o.Name == logging.OutputNameDefault || security.HasTLSCertAndKey(secret) {
```

Comment:
Copy / Paste?   I don't believe we care about the name of the output in this case for splunk

Reply:
+1 where does this special logic for default output come from? I don't think we should ever be configuring an output differently based on whether its called "default" or not. The operator is supposed to generate a default config that can be used normally.

Reply:
it was copy/past error, want to try something faster